sdk/src/Services/DynamoDBv2/Custom/DocumentModel/Document.cs

```diff
@@ -276,7 +309,37 @@ internal static DynamoDBEntry DateTimeToEpochSeconds(DynamoDBEntry entry, string
             return entry;
         }
 
-        internal static Document FromAttributeMap(Dictionary<string, AttributeValue> data, IEnumerable<string> epochAttributes)
+        // Converts a user-supplied DateTime-convertible DynamoDBEntry to epoch seconds stored in a Numeric Primitive.
+        // Differs from DateTimeToEpochSeconds by supporting dates after 2038.
+        internal static DynamoDBEntry DateTimeToEpochSecondsLong(DynamoDBEntry entry, string attributeName)
+        {
+            string epochSecondsAsString = null;
+            try
+            {
+                var dateTime = entry.AsDateTime();
+                epochSecondsAsString = AWSSDKUtils.ConvertToUnixEpochSecondsString(dateTime);
+            }
+            catch (Exception e)
+            {
+                var logger = Logger.GetLogger(typeof(Document));
+                logger.InfoFormat(
+                    "Encountered error attempting to convert '{0}' with value '{1}' to epoch seconds: {1}",
+                    attributeName, entry, e);
+
+                // To prevent the data from being (de)serialized incorrectly, we choose to throw the exception instead
+                // of swallowing it.
+                throw;
+            }
+
+            if (epochSecondsAsString != null)
```

Comment:
This can never be false anymore since we are always throwing an exception if we fail to convert. To avoid the code getting refactored in weird ways in the future I would add line 336 right after the `ConvertToUnixEpochSecondsString` call and then return. That would leave no code after the catch block and we don't need the extra if check.

Reply:
Done. Please review.